File: php/src/Domain/Country/Currency.php
<?php

namespace Katalogs\BookInvoicing\Domain\Country;

enum Currency: string
{
    case Dollar = 'USD';
    case AustralianDollar = 'AUD';
    case Euro = 'EUR';
    case PoundSterling = 'GBP';
    case Yen = 'JPY';
    case Renminbi = 'CNY';
}

File: go/domain/country/currency.go
package country

type Currency int

const (
	Dollar Currency = iota
	AustralianDollar
	Euro
	PoundSterling
	Yen
	Renminbi
)

func (c Currency) String() string {
	return [...]string{
		"Dollar",
		"AustralianDollar",
		"Euro",
		"PoundSterling",
		"Yen",
		"Renminbi",
	}[c]
}
